changeset/commit.go

```diff
@@ -17,113 +17,45 @@ limitations under the License.
 package changeset
 
 import (
-	"bufio"
+	"errors"
 	"fmt"
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"regexp"
-	"strings"
-)
-
-const (
-	commitIDFile      = "HEAD"
-	koDataPathEnvName = "KO_DATA_PATH"
-	// packedRefsFile is a file containing a list of refs, used to compact the
-	// list of refs instead of storing them on the filesystem directly.
-	// See https://git-scm.com/docs/git-pack-refs
-	packedRefsFile = "packed-refs"
+	"runtime/debug"
+	"strconv"
 )
 
 var commitIDRE = regexp.MustCompile(`^[a-f0-9]{40}$`)
+var readBuildInfo = debug.ReadBuildInfo
 
-// Get tries to fetch the first 7 digitals of GitHub commit ID from HEAD file in
-// KO_DATA_PATH. If it fails, it returns the error it gets.
+// Get returns the 'vcs.revision' property from the embedded build information
+// This function will return an error if value is not a valid Git SHA
+//
+// The result will have a '-dirty' suffix if the workspace was not clean
 func Get() (string, error) {
-	data, err := readFileFromKoData(commitIDFile)
-	if err != nil {
-		return "", err
+	info, ok := readBuildInfo()
```

Comment:
Might be worth wrapping this in a `sync.Once` since this will likely get called lots of times by a long-running controller, and the result won't change after the first time.